File: Go/Solutions/projeuler/Problem048.go
package euler

import (
	"fmt"
	"math/big"
	"time"
)

/*
Problem048 answers the problem at : https://projecteuler.net/problem=48

* Problem 48:

Find the last ten digits of the series, 1^1 + 2^2 + 3^3 + ... + 1000^1000.
*/
func Problem048() {

	start := time.Now()
	ans := big.NewInt(0)

	for i := 1; i < 1001; i++ {
		k := big.NewInt(int64(i))
		k.Exp(k, k, nil)
		ans.Add(ans, k)
	}
	ans.Mod(ans, big.NewInt(10000000000))

	end := time.Now()
	fmt.Printf("\nAnswer to Problem 42 : %d\n", ans)
	fmt.Printf("Time Taken: %f seconds\n\n", end.Sub(start).Seconds())
}

File: Python/Problem048.py
from time import time

''' 
Problem 48: https://projecteuler.net/problem=48

The series, 1**1 + 2**2 + 3**3 + ... + 10**10 = 10405071317.

Find the last ten digits of the series, 1**1 + 2**2 + 3**3 + ... + 1000**1000.
'''

start = time()
ans = sum([i**i for i in range(1, 1000)])%(10**10)

if __name__=="__main__":    
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")
